android/java/org/chromium/chrome/browser/crypto_wallet/fragments/UnlockWalletFragment.java

```diff
@@ -93,32 +93,35 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
                     }
                 });
 
-        mUnlockWalletRestoreButton.setOnClickListener(v -> {
-            if (mOnNextPage != null) {
-                mOnNextPage.gotoRestorePage(false);
-                mUnlockWalletPassword.getText().clear();
-            }
-        });
+        mUnlockWalletRestoreButton.setOnClickListener(
+                v -> {
+                    if (mOnNextPage != null) {
+                        mOnNextPage.gotoRestorePage(false);
+                        mUnlockWalletPassword.getText().clear();
+                    }
+                });
 
-        mBiometricUnlockWalletImage.setOnClickListener(v -> {
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P
-                    && Utils.isBiometricAvailable(getContext())) {
-                createBiometricPrompt();
-            }
-        });
+        mBiometricUnlockWalletImage.setOnClickListener(
+                v -> {
+                    if (Utils.isBiometricAvailable(requireContext())) {
+                        // noinspection NewApi
+                        createBiometricPrompt();
+                    }
+                });
 
-        if (mOnNextPage != null && mOnNextPage.showBiometricPrompt()) {
-            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.P
-                    || !KeystoreHelper.shouldUseBiometricOnUnlock()
-                    || !Utils.isBiometricAvailable(getContext())) {
-                showPasswordRelatedControls();
+        if (mOnNextPage != null) {
+            if (mOnNextPage.showBiometricPrompt()) {
+                if (KeystoreHelper.shouldUseBiometricOnUnlock()
+                        && Utils.isBiometricAvailable(requireContext())) {
+                    // noinspection NewApi
+                    createBiometricPrompt();
+                } else {
+                    showPasswordRelatedControls();
+                }
             } else {
-                createBiometricPrompt();
-
+                mOnNextPage.enableBiometricPrompt();
+                showPasswordRelatedControls();
             }
-        } else if (mOnNextPage != null) {
-            mOnNextPage.enableBiometricPrompt();
-            showPasswordRelatedControls();
```

Comment:
This logic was left untouched, I've just refactored `if` statements so the booleans are more readable.